schema.py

```diff
@@ -274,10 +273,17 @@ def _priority(s):
 
 
 def _invoke_with_optional_kwargs(f, **kwargs):
-    s = inspect.signature(f)
-    if len(s.parameters) == 0:
-        return f()
-    return f(**kwargs)
+    try:
+        return f(**kwargs)
+    except TypeError as e:
+        if (
+            e.args[0].startswith("{}() got an unexpected keyword argument".format(f.__name__))  # Python 2/3.
+            or e.args[0].startswith("{}() takes no arguments".format(f.__name__))  # Python 2 only.
+        ):
+            return f()
```

Comment:
Is there a realistic scenario wherein someone should be passing `**kwargs` if the function does not support keyword-arguments in the first place? 
Why not just skip the whole `try/except` block and unconditionally return `f(**kwargs)`?

Reply:
E.g. see `test_optional_callable_default_ignore_inherited_schema_validate_kwargs`: https://github.com/keleshev/schema/blob/09c00eda9599e53f7e6b84d7c91ecd3b42f71772/test_schema.py#L758-L770 (added in #268).

That approach would work if `lambda: 42` was instead `lambda **_: 42`. This would be a backwards-incompatible change (warranting a major version bump), though.